### PR TITLE
ux: change styling for exports tab on the landing page

### LINKED ISSUE #901: Improve landing page export formats UX
## Description
The export formats UX is not ideal because there are too many `JSON` words, not clear what are the export formats. The UX should be improved.

## Screenshots (if applicable)
Screenshot 2021-06-11 at 16 31 33

### PR DESCRIPTION
**Pull request by mitsosf**
closes #901 
[EDIT] closes https://github.com/inveniosoftware/invenio-app-rdm/issues/914
Added separators between export methods

image

### CODE REVIEW

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/landing_page.less [hunk 1]
```diff
@@ -210,3 +210,12 @@
...
     float: right;
 }
+
+.exports-list > div.item:first-child {
+  padding-left: 0!important;
+  border-left: none!important;
+}
+
+.exports-list > div.item:last-child {
+  border-right: none!important;
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/landing_page.less:220]:
```suggestion
  padding-right: 0!important;
  border-right: none!important;
```

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/landing_page.less [hunk 2]
```diff
@@ -210,3 +210,12 @@
     display: inline-block;
     float: right;
 }
+
+.exports-list > div.item:first-child {
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/landing_page.less:214]: I was wondering, if you add the class to the `Semantic UI` theme, [here](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/elements/list.overrides#L39), can you avoid the `!important`?
See the other examples there...

**Comment by ntarocco** [comment]
Just a reminder for reviewers: missing to apply the same style change to the citations string list.
@mitsosf be aware that the citations string list is a React application, [here](https://github.com/mitsosf/invenio-app-rdm/blob/master/invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js#L94). You will have to rebase from `master`.

**Review by ppanero — COMMENTED** [review]
One more tiny thing on the commit message. It is good, but we *try* (not always manage but...) to follow the [Invenio convention](https://invenio.readthedocs.io/en/latest/community/contributing/contribution-guide.html#commit-messages). In your case I think is just a matter of re-oder.. something like:

```
 landing page: change styling for exports tab (#901)
```

**Comment by mitsosf** [comment]
Here we can see the copy button and the citation styles separated as the ones above:

image

**Review by jrcastro2 — APPROVED** [review]
LGTM!
It needs a rebase.
I usually prefer to have the "copy to clipboard" button on the right and a bit smaller, like this:
image
WDYT?

Also ping @lnielsen or @ntarocco since we are adding a new npm package (`react-copy-to-clipboard`) and I am not sure what are the politics around this (if there are any)

**Comment by mitsosf** [comment]
Will rebase.

I also prefer the button on the right, the main issue was if I put it in a fixed column at the end, it would show weird if the title is very short (essentially "pinned to the right").

If I didn't put it in a fixed column, the button would show after the text, which was an issue if the text was very long (e.g 10 lines).

The way I implemented it now (on the left), at least it stays at a fixed spot and while I didn't like the design at first, I got used to it so decided to leave it as it is.

Regarding the new package, @ntarocco recommended this implementation from a different project we had already used it in, so I guess it's ok to re-use it here as well.

**Comment by ppanero** [comment]
I understand the issue with the clipboard on the right... Would i be to weird or "far" from the content to put it next to the `Citation` title? Otherwise, maybe boxing the citation itself? e.g. docker:

image

**Comment by ntarocco** [comment]
> react-copy-to-clipboard

[react-copy-to-clipboard](https://www.npmjs.com/package/react-copy-to-clipboard) has MIT license, good to go :)

**Comment by ntarocco** [comment]
Proposing something else, even if I don't love it...

124733325-61ea0a00-df14-11eb-9ee8-3c55fddc617b

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utlis/CopyButton.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  render() {
+    const { text, onCopy } = this.props;
+    return (
+      <CopyToClipboard
+        text={text}
+        onCopy={() => {
+          onCopy(text);
+        }}
+      >
+        <Button className="copy" basic icon="copy" />
```
Comment (jrcastro2) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utlis/CopyButton.js:22]:
```suggestion
        <Button size="mini" className="copy" basic icon="copy" />
```

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js [hunk 4]
```diff
@@ -84,32 +87,48 @@ export class RecordCitationField extends Component {
...
+        <div className="row">
+          <div className="ui grid">
+            <div className="one wide column">
+              {<CopyButton text={citation} position='top'/>}
+            </div>
+            <div className="fifteen wide column">
+              <div id="citation-text">
+                {loading ? this.placeholderLoader() : citation}
+              </div>
+            </div>
```
Comment (jrcastro2) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js:128]: An alternative:
```suggestion
            <div className="sixteen wide column">
              <div id="citation-text">
                {loading ? this.placeholderLoader() : citation}
                {<CopyButton text={citation} position='top'/>}
              </div>
            </div>
```

**Review by jrcastro2 — COMMENTED** [review]
## Another alternative
With the suggested changes below ti would look like this:
### Short titles 
image

### Long titles
image

It's missing some left margin but can be added in the button class.
WDYT?

**Comment by lnielsen** [comment]
Hmm.....I would keep the copy button on the right side as Javier has suggested next to the text that it copies........that said, I don't feel strongly about it.

**Comment by lnielsen** [comment]
Let's take a decision during the standup today.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utlis/CopyButton.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
+// This file is part of InvenioRDM
```
Comment (ppanero) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utlis/CopyButton.js:1]: comment not on the content, but on the package itself. There was already a utils.js file, but this is creating a folder.... shall we try to put all these "generic things in more meaningfull named packages/folder? e.g. we have `EditButton` in the langind page, but this goes to utils... Shall we create a `buttons` folder... random thought on code organization hehe

Comment (ntarocco) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utlis/CopyButton.js:1]: Yes please! No utils packages/files whenever possible!
The CopyBtn cmp might go in the future in some common lib to be reusable...

**Comment by jrcastro2** [comment]
### Another alternative
Copy it by clicking on it:
gif

**Comment by lnielsen** [comment]
From standup:
- Right aligned button (docker/GitHub style)

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utlis/CopyButton.js (hunk 5 above)
Comment (mitsosf) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utlis/CopyButton.js:1]: Yes, I will change the directory name (which  was actually misspelled :upside_down_face:)  to `buttons`

**Comment by mitsosf** [comment]
> From standup:
> 
> * Right aligned button (docker/GitHub style)

Meaning in a "box" or just stuck to the right?

**Comment by zzacharo** [comment]
Screenshot 2021-07-09 at 13 52 15

**Pull request merged**